SpatialGDK/Source/SpatialGDKServices/Private/SSpatialOutputLog.cpp

```diff
@@ -255,6 +264,40 @@ void SSpatialOutputLog::FormatAndPrintRawLogLine(const FString& LogLine)
 	// Log categories take the form "improbable.deployment.InternalGameLauncher", we filter to the last category to make it more human readable.
 	LogCategory.Split(TEXT("."), nullptr, &LogCategory, ESearchCase::IgnoreCase, ESearchDir::FromEnd);
 
+	// Remove trailing whitespace.
+	LogCategory = LogCategory.TrimEnd();
+
+	// For worker logs 'WorkerLogMessageHandler' we use the worker name as the category. The worker name can be found in the msg.
+	// msg=[WORKER_NAME:TYPE] ... e.g. msg=[UnrealWorkerF6DD366E460D1080061C2D88FFA08C1F:Unreal]
+	if (LogCategory == TEXT("WorkerLogMessageHandler"))
+	{
+		FString WorkerNameAndType;
+		FString NewLogMessage;
+
+		// We split at the closing square brace of the log message.
+		LogMessage.Split(TEXT("]"), &WorkerNameAndType, &NewLogMessage);
+
+		// Now remove the ':Unreal' tag
+		FString WorkerName;
+		FString WorkerType;
+		WorkerNameAndType.Split(TEXT(":"), &WorkerName, &WorkerType);
+
+		// Remove the remaining '[' at the start of the WORKER_NAME.
+		WorkerName.RemoveAt(0);
+
+		// Shorten the hash at the end of the WorkerName for Unreal workers.
+		if (WorkerType.Contains("Unreal"))
```

Comment:
```suggestion
		if (WorkerType.Contains(TEXT("Unreal")))
```

Reply:
Is it intentional that this ignores the existence of AIWorker, CrashBotWorker, etc (i.e. for offloading)?

Reply:
Good spot, You can get a set of current worker types from: https://github.com/spatialos/UnrealGDK/blob/358a5ef8a4b2992999eb05e2a3782dc158b425ac/SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h#L199

Reply:
So to make this correct I've had to do a bit of digging and rejigging to correctly identify an editor worker for hash shortening.

Turns out the `WorkerType` is set in the creation of the `SpatialOutputDevice`, this was hardcoded to "Unreal". In order to correctly identify editor and non-editor workers I've created a constant in `SpatialConstants` ("Unreal") for non-editor and a new constant file `SpatialGDKServicesConstants` ("UnrealEditor") for editor workers. This is a little bit cumbersome but it's because the `SpatialGDKServicesModule` where this code lives cannot have a dependency on the `SpatialGDK` as this would cause a circular dependency and break Mac and iOS builds... Additionally I'd rather have the constants than hardcoding for maintainability. I'm open to suggestions here though.